functions/_fifc.fish

```diff
@@ -49,7 +49,13 @@ function _fifc
     # We use eval hack because wrapping source command
     # inside a function cause some delay before fzf to show up
     eval $cmd | while read -l token
-        set -a result (string escape -- $token)
+        # string escape will escape '~' if present (at the begenning of path).
+        # so we need to exclude it from escaping
+        if string match --quiet '~*' -- $token
+            set -a result (string join -- "" "~" (string sub --start 2 -- $token | string escape))
```

Comment:
This fix accounts for 99% use cases, but it doesn't completely fix the problem, for example `~root` won't work.